### PR TITLE
added support for the x-api-user-agent header and the partner reference

### PR DESCRIPTION
**Pull request by wvdiepen**

### CODE REVIEW

**Comment by mvanzelst** [comment]
Can we add a new field to the payment info block in the example webshop (src/main/webapp/WEB-INF/templates/order_details.ftl) for the user agent and partner reference field

**Comment by mvanzelst** [comment]
Can you add the user agent setting to the example webshop as well? The endpoint is initialised in the constructor of nl.rabobank.gict.payments_savings.omnikassa_frontend.test.webshop.WebshopController#WebshopController

File: sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java [hunk 1]
```diff
@@ -46,13 +46,37 @@ public final class Endpoint {
...
      * @param signingKey    this is the key given by the Rabobank to sign all communication
      * @param tokenProvider this must be your own implementation of the tokenProvider, see developer-manual
-     * @deprecated This method is deprecated, use the {@link Endpoint#createInstance(Environment, String, TokenProvider)} instead.
      * @return new instance of Endpoint
      * Note: for backward compatibility do not remove this method!
+     * @deprecated This method is deprecated, use the {@link Endpoint#createInstance(Environment, String, TokenProvider)} instead.
      */
     @Deprecated
     public static Endpoint createInstance(String baseURL, byte[] signingKey, TokenProvider tokenProvider) {
         ApiConnector connector = new ApiConnector(baseURL, signingKey);
```
Comment (mvanzelst) [inline sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java:55]: This line can be removed

File: sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java [hunk 2]
```diff
@@ -67,8 +91,42 @@ public static Endpoint createInstance(String baseURL, byte[] signingKey, TokenPr
...
+    /**
+     * Use this method to retrieve an instance of the Endpont
+     *
+     * @param environment             this is the Url that points to the Rabobank API
+     * @param base64encodedSigningKey this is the key given by the Rabobank to sign all communication
+     * @param tokenProvider           this must be your own implementation of the tokenProvider, see developer-manual
+     * @param userAgent               this is the User-Agent value you want to give your implementation
+     * @return new instance of Endpoint
+     */
+    public static Endpoint createInstance(Environment environment,
```
Comment (mvanzelst) [inline sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java:106]: I think providing a constructor with both the userAgent and partner reference (the one below) is enough. This one can be removed

File: sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java [hunk 3]
```diff
@@ -46,13 +46,37 @@ public final class Endpoint {
...
+     * @param signingKey       this is the key given by the Rabobank to sign all communication
+     * @param tokenProvider    this must be your own implementation of the tokenProvider, see developer-manual
+     * @param userAgent        this is the User-Agent value you want to give your implementation
+     * @param partnerReference this can be filled with the partner reference, if applicable.
+     * @return new instance of Endpoint
+     * Note: for backward compatibility do not remove this method!
+     * @deprecated This method is deprecated, use the {@link Endpoint#createInstance(Environment, String, TokenProvider)} instead.
+     */
+    @Deprecated
+    public static Endpoint createInstance(String baseURL,
```
Comment (mvanzelst) [inline sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java:72]: Are we sure we need to add this to the deprecated factory method as well? I see our test shop uses this factory method. Can't it be migrated to the new factory method?

Comment (wvdiepen) [reply sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java:72]: No we can't change this one, then we cannot test locally anymore

**Review by mvanzelst — APPROVED** [review]

**Pull request merged**